runtime/workerpool/workerpool.go

```diff
@@ -93,6 +94,28 @@ func (w *WorkerPool) Submit(workerFunc func(), optStackTrace ...string) {
 	w.Queue.Push(newTask(workerFunc, w.decreasePendingTasks, lo.First(optStackTrace)))
 }
 
+// DebounceFunc returns a function that can be used to submit a task that is canceled if the function is called with a
+// new task before the previous task was executed.
+func (w *WorkerPool) DebounceFunc() (debounce func(workerFunc func(), optStackTrace ...string)) {
+	var lastInvocation atomic.Uint64
+	var execMutex sync.Mutex
+
+	return func(workerFunc func(), optStackTrace ...string) {
+		currentInvocation := lastInvocation.Add(1)
```

Comment:
Please add some comments in the code to explain the flow a little. It took me a moment to understand the flow.

Reply:
Done